OneCommunityGlobal/HighestGoodNetworkApp
# Roberto - alters returnUpdatedBadgesCollection function to take into account duplicate badges

# Description
This Pr is meant to handle this instance : [Duplicate badge count not handled correctly](https://github.com/OneCommunityGlobal/HGNRest/pull/666#pullrequestreview-1899667634),
the linked backend pr originally was supposed to handle instances of duplicate badges and merge them successfully. Possibly some new frontend changes have changed some logic in which cases of duplicate badges' counts are already wrong before they reach the backend. The returnUpdatedBadgesCollection function would add an extra count for each instance of a duplicate to that badge and cause problems before reaching the backend which resulted in counts being higher than they were expected to be. This pr is meant to correct the logic and allow for the backend to receive the correct count and handle the rest of the process.

## Related PRS (if any):
This frontend PR is related to the #[666](https://github.com/OneCommunityGlobal/HGNRest/pull/666) frontend PR, it is an extension to handle a minor edge case on frontend.
Use the 666 backend pr with this ^^.

## Main changes explained:
- alters returnUpdatedBadgesCollection function to take into account duplicate badges, due to some new frontend changes if the returnUpdatedBadgesCollection function receives duplicate badges, it does not know how to handle the instance and adds to the badge count more instances than it should. The changes here allow for the count to be added correctly, the paired backend handles the rest and fully merges the duplicates.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. follow backend instructions:[backend]( https://github.com/OneCommunityGlobal/HGNRest/pull/666)
